### ISSUE TITLE
OSRM coverage for Trinidad and Tobago

### ISSUE THREAD

**Issue by vovkind** [Project-OSRM/osrm-backend#5979]
Hi all.

Can someone tell where can I find OSRM data for this country?

It is not listed under https://download.geofabrik.de/central-america.html

Thanks a lot.

**Comment by frodrigo** [comment]
OSM data are available, not OSRM ones. You have to process OSM to get OSRM.

Geofabrik server are down for now, but will be back. You can try OSM-FR http://download.openstreetmap.fr/extracts/

**Comment by frodrigo** [comment]
One extract is available here
http://download.openstreetmap.fr/extracts/central-america/

If the extract is not on Geofabrk look at OSM-FR ;-), we made all countries and some sub countries for the need of Osmose-QA.

**Comment by alex131290** [comment]
When using this file http://download.openstreetmap.fr/extracts/central-america/trinidad_and_tobago-latest.osm.pbf 
The osrm_preprocessing of Trinidad is failing:

Here's the error:

`what(): There are no edges remaining after parsing.src/extractor/extractor.cpp:645`

Here's the full log:

`2021-03-21T08:23:03Z | INFO  | Ensure local pbf dir /tmp/pbf exists
2021-03-21T08:23:03Z | INFO  | Downloading OSM data from S3 path s3://dragontail-openstreetmaps-prod/prod/source_pbfs/trinidad_and_tobago-latest.osm.pbf
Completed 256.0 KiB/349.5 KiB (3.4 MiB/s) with 1 file(s) remaining
Completed 349.5 KiB/349.5 KiB (4.6 MiB/s) with 1 file(s) remaining
download: s3://dragontail-openstreetmaps-prod/prod/source_pbfs/trinidad_and_tobago-latest.osm.pbf to ../tmp/pbf/data.osm.pbf
2021-03-21T08:23:03Z | INFO  | Extracting pbf
[info] Parsed 0 location-dependent features with 0 GeoJSON polygons
[info] Using script /usr/local/share/osrm/profiles/car.lua
[info] Input file: data.osm.pbf
[info] Profile: car.lua
[info] Threads: 16
[info] Parsing in progress..
[info] input file generated by pyosmium-up-to-date/3.0.1
[info] timestamp: 2021-03-20T08:48:58Z
[info] Using profile api version 4
[info] Found 3 turn restriction tags:
[info]   motorcar
[info]   motor_vehicle
[info]   vehicle
[info] Parse relations ...
[info] Parse ways and nodes ...
[info] Using profile api version 4
[info] Using profile api version 4
[info] Parsing finished after 0.073649 seconds
[info] Raw input contains 53 nodes, 1 ways, and 348 relations, 20 restrictions
terminate called after throwing an instance of 'osrm::util::exception'
  what():  There are no edges remaining after parsing.src/extractor/extractor.cpp:645
Command terminated by signal 6
	Command being timed: "osrm-extract -p /usr/local/share/osrm/profiles/car.lua /tmp/pbf/data.osm.pbf"
	User time (seconds): 0.08
	System time (seconds): 0.05
	Percent of CPU this job got: 81%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 0:00.17
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 44028
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 10608
	Voluntary context switches: 427
	Involuntary context switches: 86
	Swaps: 0
	File system inputs: 0
	File system outputs: 68672
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0
2021-03-21T08:23:04Z | ERROR | Failed to extract pbf
`

Here's another issue that i found related to the error I'm getting when trying to process it - https://github.com/Project-OSRM/osrm-backend/issues/5589

Would it be possible to upload/generate the Trinidad.pbf file in the same way it was generated for the rest of countries available here https://download.geofabrik.de/

**Comment by frodrigo** [comment]
No these pbf come from different provider with different method to produce theme. These pdf are outside the scope of the OSRM project.

Nevertheless I oppened the trinidad_and_tobago-latest.osm.pbf extract and it is bugged. It contains only one marine cable.

I opened an issue here https://github.com/osm-fr/infrastructure/issues/280

Unless you are really interested by Trinidad, try an other extract.
